src/isotools/_utils.py

```diff
@@ -561,3 +561,135 @@ def cmp_dist(a, b, min_dist=3):
+            if exon_chain not in cov:
+                cov[exon_chain] = 0
+            for s, n in tr[field].items():
+                if s in samples:
+                    cov[exon_chain] += n
+            if cov[exon_chain] == 0:
+                del cov[exon_chain]
+    else:
+        field = f'{feature}_unified'
+        for tr in transcripts:
+            if tr[field] is None:
+                continue
+
+            for s,pos_dict in tr[field].items():
+                if s in samples:
+                    for pos, n in pos_dict.items():
+                        if pos not in cov:
+                            cov[pos] = 0
+                        cov[pos] += n
+
+    # keep ones with coverage > 0
+    cov = {k: v for k, v in cov.items() if v > 0}
+
+    if len(cov) == 0:
+        return [], []
+
+    # sort the coverage in descending order and return
+    abundance, occurrence = zip(*sorted(zip(list(cov.values()), list(cov.keys())), reverse=True))
+    return list(abundance), list(occurrence)
+
+
+def count_distinct_pos(pos_list, strict_pos=15):
+    '''
+    :param pos_list: A list of TSS/PAS positions, sorted by their abundance descendingly (output from structure_feature_cov).
+    :param strict_pos: Difference allowed between two positions when considering identical TSS/PAS.
+    :return: How many distinct positions are there.
+    '''
+
+    merged_idx = []
+    for x in range(len(pos_list)-1):
+        if x in merged_idx:
+            continue
+        for y in range(x+1, len(pos_list)):
+            if y in merged_idx:
+                continue
+            if pos_list[x] - strict_pos <= pos_list[y] <= pos_list[x] + strict_pos:
+                #print(f'{x}:{pos_list[x]} and {y}: {pos_list[y]} are close enough')
+                # keep the one with higher coverage
+                merged_idx.append(y)
+    return len(pos_list) - len(merged_idx)
+
+
+def count_distinct_exon_chain(ec_list, strict_ec=0, strict_pos=15):
+    '''
+    :param ec_list: A list of exon chains, sorted by their abundance descendingly (output from structure_feature_cov).
+    :param strict_ec: Distance allowed between each position, except for the first/last, in two exon chains so that they cab be considered as identical.
+    :param strict_pos: Difference allowed between two positions when considering identical TSS/PAS.
+    :return: How many distinct exon chains are there.
+    '''
+
+    merged_idx = []
+    for x in range(len(ec_list)-1):
+        if x in merged_idx:
+            continue
+        for y in range(x+1, len(ec_list)):
+            if y in merged_idx:
+                continue
+            # if the number of exons is different, skip
+            if len(ec_list[x]) != len(ec_list[y]):
+                continue
+
+            pos_in_x = [pos for exon in ec_list[x] for pos in exon]
+            pos_in_y = [pos for exon in ec_list[y] for pos in exon]
+
+            pos_diff = [abs(x - y) for x,y in zip(pos_in_x, pos_in_y)]
```

Comment:
Please don't use x and y here. Initial x and y are indices. These here are genomic positions